### ISSUE TITLE
test_reproject_healpix_to_image_footprint failures on Debian with ARM (and other) CPUs

### ISSUE THREAD

**Issue by olebole** [astropy/reproject#414]
On the Debian architectures with ARM (32+64 bit), RISCv64, all **test_reproject_healpix_to_image_footprint(…, 'nearest-neighbor')** test fail:
```
_ test_reproject_healpix_to_image_footprint[1-True-C-C-dtype1-nearest-neighbor] _

nside = 1, nested = True, healpix_system = 'C', image_system = 'C'
dtype = dtype('uint8'), order = 'nearest-neighbor'

    @pytest.mark.parametrize(
        "nside,nested,healpix_system,image_system,dtype,order",
        itertools.product(
            [1, 2, 4, 8, 16, 32, 64],
            [True, False],
            "C",
            "C",
            ALL_DTYPES,
            ["bilinear", "nearest-neighbor"],
        ),
    )
    def test_reproject_healpix_to_image_footprint(
        nside, nested, healpix_system, image_system, dtype, order
    ):
        """Test that HEALPix->WCS conversion correctly flags pixels that do not
        have valid WCS coordinates."""
    
        npix = nside_to_npix(nside)
        healpix_data = np.random.uniform(size=npix).astype(dtype)
    
        reference_header = get_reference_header(overscan=2, oversample=2, nside=nside)
    
        wcs_out = WCS(reference_header)
        shape_out = reference_header["NAXIS2"], reference_header["NAXIS1"]
    
        image_data, footprint = reproject_from_healpix(
            (healpix_data, healpix_system),
            wcs_out,
            shape_out=shape_out,
            order=order,
            nested=nested,
        )
    
        if order == "bilinear":
            expected_footprint = ~np.isnan(image_data)
        else:
            coord_system_in = parse_coord_system(healpix_system)
            yinds, xinds = np.indices(shape_out)
            world_in = wcs_out.pixel_to_world(xinds, yinds).transform_to(coord_system_in)
            world_in_unitsph = world_in.represent_as("unitspherical")
            lon_in, lat_in = world_in_unitsph.lon, world_in_unitsph.lat
            expected_footprint = ~(np.isnan(lon_in) | np.isnan(lat_in))
    
>       np.testing.assert_array_equal(footprint, expected_footprint)

reproject/healpix/tests/test_healpix.py:86: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<built-in function eq>, array([[1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
        1., 1., 1., 1...e,
        False, False, False, False, False, False, False, False, False,
        False, False, False, False, False]]))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 359 / 512 (70.1%)
E           Max absolute difference: 1.
E           Max relative difference: 0.
E            x: array([[1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
E                   1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.],
E                  [1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,...
E            y: array([[False, False, False, False, False, False, False, False, False,
E                   False, False, False, False, False, False, False, False, False,
E                   False, False, False, False, False, False, False, False, False,...

/usr/lib/python3.11/contextlib.py:81: AssertionError
```

The ones with **order='bilinear'** pass.
[Full build log for amd64](https://buildd.debian.org/status/fetch.php?pkg=reproject&arch=arm64&ver=0.13.0-1&stamp=1701005789&raw=0).
In Intel (amd64, i386), PowerPC, s390x, all tests pass.

Environment:

* Python 3.11.6
* Numpy 1.24.2
* Scipy: 1.10.1
* Astropy 5.3.4
* Astropy-healpix 0.6

**Comment by lpsinger** [comment]
Hmmmm. This comes to mind: https://github.com/astropy/astropy-healpix/blob/a746aac91b4790aae8ed9eafa85262f78ce8f5ab/astropy_healpix/_core.c#L8-L13

**Comment by lpsinger** [comment]
@olebole, we think that https://github.com/astropy/astropy-healpix/pull/208 may have fixed this. We'll roll another release of astropy-healpix shortly.

**Comment by lpsinger** [comment]
@olebole, we think that this should be fixed now if you updated to astropy-healpix 1.0.1.

**Comment by olebole** [comment]
Yes, I can confirm this. After uploading healpix 1.0.1, the reproject tests on all platforms pass.
